src/types/index.ts

```diff
@@ -1,8 +1,36 @@
+import {Bucket, Cell, Dashboard, Task} from "../api";
+import { Label as APILabel } from "../api";
+
+interface ILabelProperties {
+  color: string;
+  description?: string;
+}
+
+export interface ILabel extends APILabel {
+  properties: ILabelProperties;
+}
+
 export interface ISetupParams {
   username: string;
   password: string;
   org: string;
   bucket: string;
 }
 
+export interface IBucket extends Bucket {
+  labels: ILabel[];
+}
+
+export interface ITask extends Task {
+  labels: ILabel[];
+}
+
+export interface IDashboard extends Dashboard {
+  orgID: string;
+  id: string;
+  name: string;
+  labels: ILabel[];
+  cells: Cell[];
```

Comment:
I'm pretty worried here that we are creating brand new properties here that are decoupled from the swagger definition, just to make them required. If swagger changes `orgID` to `someTerribleParamaterName` we will still be enforcing `orgID` here, and our types will still have them. if swagger changes orgID from string to number type, we are essentially overwriting that here. I don't have a quick fix for mapping over a subset of properties, But mapped types in typescript provide a good way to convert optional properties to required ones. One kind of convoluted solution is creating an intermediate interface which picks from Dashboard, and mapping over it to make the properties required and then extend Dashboard with it.